tests/Endatix.Infrastructure.Tests/Identity/Authentication/JwtTokenServiceTests.cs

```diff
@@ -0,0 +1,251 @@
+        result.Should().NotBeNull();
+        result.Token.Should().NotBeNullOrEmpty();
+        result.ExpireAt.Should().BeAfter(DateTime.UtcNow);
+
+        var handler = new JwtSecurityTokenHandler();
+        var jsonToken = handler.ReadToken(result.Token) as JwtSecurityToken;
+        jsonToken.Should().NotBeNull();
+        jsonToken!.Claims.Should().Contain(c => c.Type == JwtRegisteredClaimNames.Sub && c.Value == "1");
+        jsonToken.Claims.Should().Contain(c => c.Type == JwtRegisteredClaimNames.Email && c.Value == "test@example.com");
+        jsonToken.Claims.Should().Contain(c => c.Type == JwtRegisteredClaimNames.NameId && c.Value == "testuser");
+        jsonToken.Claims.Should().Contain(c => c.Type == ClaimNames.Role && c.Value == RoleNames.ADMIN);
+        jsonToken.Claims.Should().Contain(c => c.Type == ClaimNames.Permission && c.Value == Allow.AllowAll);
+    }
+
+    [Fact]
+    public void ValidateAccessToken_ValidToken_ReturnsSuccessWithUserId()
+    {
+        // Arrange
+        var optionsWrapper = Substitute.For<IOptions<JwtOptions>>();
+        optionsWrapper.Value.Returns(_validJwtOptions);
+        var tokenService = new JwtTokenService(optionsWrapper);
+        var user = new User(1, "testuser", "test@example.com", false);
+        var token = tokenService.IssueAccessToken(user);
+
+        // Act
+        var result = tokenService.ValidateAccessToken(token.Token);
+
+        // Assert
+        result.IsSuccess.Should().BeTrue();
+        result.Value.Should().Be(1);
+    }
+
+    [Fact]
+    public void ValidateAccessToken_InvalidToken_ReturnsInvalidResult()
+    {
+        // Arrange
+        var optionsWrapper = Substitute.For<IOptions<JwtOptions>>();
+        optionsWrapper.Value.Returns(_validJwtOptions);
+        var tokenService = new JwtTokenService(optionsWrapper);
+
+        // Act
+        var result = tokenService.ValidateAccessToken("invalidToken");
+
+        // Assert
+        result.IsSuccess.Should().BeFalse();
+        result.ValidationErrors.Should().NotBeNull();
+        result.ValidationErrors.First().ErrorMessage.Should().Contain("JWT must have three segments (JWS) or five segments (JWE).");
+    }
+
+    [Fact]
+    public void IssueRefreshToken_Always_ReturnsValidToken()
+    {
+        // Arrange
+        var optionsWrapper = Substitute.For<IOptions<JwtOptions>>();
+        optionsWrapper.Value.Returns(_validJwtOptions);
+        var tokenService = new JwtTokenService(optionsWrapper);
+
+        // Act
+        var result = tokenService.IssueRefreshToken();
+
+        // Assert
+        result.Should().NotBeNull();
+        result.Token.Should().NotBeNullOrEmpty();
+        result.ExpireAt.Should().BeAfter(DateTime.UtcNow.AddDays(_validJwtOptions.RefreshExpiryInDays - 1));
+        result.ExpireAt.Should().BeBefore(DateTime.UtcNow.AddDays(_validJwtOptions.RefreshExpiryInDays + 1));
+    }
+
+    [Fact]
+    public async Task RevokeTokensAsync_NullUser_ReturnsNotFoundResult()
```

Comment:
@dspetrov  - do we really need tests for RevokeTokens? Maybe I am missing something, but the method is a placeholder as of now, so I think these tests will give just false negatives that there a not issues. Maybe we don't need these tests or leave them? In any case, there's no much value in them for now I think. 

Maybe a better approach will be to add implementation for the `RevokeTokensAsync` method and delete the refresh token of the user. What's you take here?

Reply:
The test passes and yes, it may not provide much value but it doesn't hurt also. It will be extended if the method has a full implementation.